### PR TITLE
fix: preview reload in sbas

### PR DESCRIPTION
**Pull request by voicis**
Livereload is served from a different url in SBAS. Changed CPE coding to correctly handle such scenarios.

### CODE REVIEW

**Review by Jimmy-Joseph19 — APPROVED** [review]
Changes looks good.
Test coverage good.
Changeset found.
Did not test manually.

File: .changeset/few-pugs-protect.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@sap-ux/control-property-editor': patch
+'@sap-ux/preview-middleware': patch
+'@sap-ux/reload-middleware': patch
+'@sap-ux/btp-utils': patch
+---
+
+fix preview reload on SBAS
```
Comment (zdravko-georgiev) [inline .changeset/few-pugs-protect.md:8]: Could you please add more information about what change we are doing and what?

File: packages/btp-utils/src/app-studio.ts [hunk 2]
```diff
@@ -84,3 +84,14 @@ export async function listDestinations(): Promise<Destinations> {
...
+
+/**
+ * Exposes port in SAP Business Application Studio.
+ *
+ * @param port Port that needs to be exposed
+ * @returns url on which the port is exposed
+ */
+export async function exposePort(port: number): Promise<string> {
+    const response = await axios.get(`http://localhost:3001/AppStudio/api/getHostByPort?port=${port}`);
+    return `${response.data.result}`;
```
Comment (zdravko-georgiev) [inline packages/btp-utils/src/app-studio.ts:96]: Please add `try/catch` for the cases when the API is not available. The exception should not block the process of starting the local server.

File: packages/btp-utils/test/app-studio.test.ts [hunk 3]
```diff
@@ -144,4 +150,19 @@ describe('App Studio', () => {
...
+
+    describe('exposePort', () => {
+        const server = 'http://localhost:3001/';
+
+        test('calls app studio api with correct port', async () => {
+            nock(server).get('/AppStudio/api/getHostByPort?port=1234').reply(200, { result: 'https://abcd.com' });
+            const url = await exposePort(1234);
+            expect(url).toStrictEqual('https://abcd.com');
+        });
+        test('calls app studio api with correct port', async () => {
```
Comment (zdravko-georgiev) [inline packages/btp-utils/test/app-studio.test.ts:162]: This should be saying `testing error case` 😉

**Review by zdravko-georgiev — COMMENTED** [review]
Just one small comment, rest looks OK

**Review by zdravko-georgiev — APPROVED** [review]
Code change looks good
Review comments addressed
Very good test coverage
Did NOT tested manually

**Review by Jimmy-Joseph19 — APPROVED** [review]
Reapproving. Did not test locally

**Pull request merged**